lib/galaxy/webapps/galaxy/api/histories.py

```diff
@@ -577,3 +579,13 @@ def get_custom_builds_metadata(self, trans, id, payload=None, **kwd):
             'installed_builds': [{'label': ins, 'value': ins} for ins in installed_builds],
             'fasta_hdas': [{'label': f'{hda.hid}: {hda.name}', 'value': trans.security.encode_id(hda.id)} for hda in fasta_hdas],
         }
+
+    @expose_api
+    def sharing(self, trans, id, payload=None, **kwd):
+        """
+        * GET/POST /api/pages/{id}/sharing
+            View/modify sharing options for the page with the given id.
```

Comment:
We should split this up into GET and POST and make the payload required for POST actions. But again, we can do that in another PR.

Reply:
Many thanks for the review!!

Yeah I agree, in the future FastAPI controller there will be two separate actions like it is already in Pages [here](https://github.com/galaxyproject/galaxy/pull/11712/files#diff-4a8e2455a0a6963c425766e46c399a63c23ecbbebc8d347514a08b3d56dbeabbR136-R160) but, since we will deprecate those legacy controllers at some point I tried to just add the bare minimum to the old ones 😅